include/misc/sem.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief public sys_sem APIs.
+ */
+
+#ifndef ZEPHYR_INCLUDE_MISC_SEM_H_
+#define ZEPHYR_INCLUDE_MISC_SEM_H_
+
+#ifdef CONFIG_USERSPACE
+/*
+ * sys_sem behaves almost exactly like k_sem, with the added
```

Comment:
Is "almost exactly" a proper expression?
@wentongwu are there any other differences with k_sem, other than the fact that sys_sem can reside in user memory?